docs/CONNECTOR_PROTOCOL.md

```diff
@@ -82,29 +110,111 @@ This is our main communication index, used to communicate the connector's config
+        "active" : {
+          "properties" : {
+            "rules" : {
+              "properties" : {
+                "id" : { "type" : "keyword" },
+                "policy" : { "type" : "keyword" },
+                "field" : { "type" : "keyword" },
+                "rule" : { "type" : "keyword" },
+                "value" : { "type" : "keyword" },
+                "order" : { "type" : "short" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "advanced_snippet" : {
+              "properties" : {
+                "value" : { "type" : "object" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "validation" : {
+              "properties" : {
+                "state" : { "type" : "keyword" },
+                "errors" : {
+                  "properties" : {
+                    "ids" : { "type" : "keyword" },
+                    "messages" : { "type" : "text" }
+                  }
+                }
+              }
+            }
+          }
+        },
+        "draft" : {
+          "properties" : {
+            "rules" : {
+              "properties" : {
+                "id" : { "type" : "keyword" },
+                "policy" : { "type" : "keyword" },
+                "field" : { "type" : "keyword" },
+                "rule" : { "type" : "keyword" },
+                "value" : { "type" : "keyword" },
+                "order" : { "type" : "short" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "advanced_snippet" : {
+              "properties" : {
+                "value" : { "type" : "object" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "validation" : {
+              "properties" : {
+                "state" : { "type" : "keyword" },
+                "errors" : {
+                  "properties" : {
+                    "ids" : { "type" : "keyword" },
+                    "messages" : { "type" : "text" }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
     "index_name" : { "type" : "keyword" },
     "is_native" : { "type" : "boolean" },
     "language" : { "type" : "keyword" },
     "last_seen" : { "type" : "date" },
-    "last_sync_error" : { "type" : "text" },
+    "last_sync_error" : { "type" : "keyword" },
     "last_sync_status" : { "type" : "keyword" },
     "last_synced" : { "type" : "date" },
-    "last_indexed_document_count" : { "type" : "integer" },
-    "last_deleted_document_count" : { "type" : "integer" },
+    "last_indexed_document_count" : { "type" : "long" },
+    "last_deleted_document_count" : { "type" : "long" },
```

Comment:
These are unfortunate changes, but are necessary because these were missing from the ent-search mapping definition. Elasticsearch therefore dynamically mapped these numeric fields as `long` in 8.5, so in order to avoid a reindex operation, we now need to explicitly define the mapping type as `long` to match.